generate-examples/generate-cert-chain.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+# this script generates a 3-cert ECDSA P-256 chain (root -> CA -> issuer) valid for 5 years
+
+# directory where intermediate files are kept
+tmpdir=certs
+mkdir -p $tmpdir
+# directory where generated JWK files are stored
+outdir=certs
+mkdir -p $outdir
+
+# generate self-signed root CA cert
+openssl req -x509 -new -newkey ec:<(openssl ecparam -name secp521r1) -keyout $tmpdir/root_CA.key -out $tmpdir/root_CA.crt -nodes -subj "/CN=SMART Health Card Example Root CA" -days 1825
+
+# generate intermediate CA cert request
```

Comment:
I wonder if it's worth also showing an "easy example" sans intermediate CA.

Reply:
What do you mean? Having a JWK with x5c containing only one self-signed cert? Not sure what the value (security wise) to use this deployment, as trusting a self-signed cert is the same as trusting a JWKS. Or are you suggesting a length 2 chain (root -> leaf cert)?

Reply:
I meant a length 2 chain (root -> leaf)

Reply:
3-cert chains are fairly common for web sites; not sure if it translate automatically for the envisioned trust frameworks, but I would assume it is a fairly common use case: trust framework root -> member CA (e.g., a hospital, pharmacy, etc.) -> point-of-service cert (clinic, local branch)). We can add a 2nd example with a 2-cert chain, but we want to make sure verifiers don't hardcode a length 2 chain cert+CA in their code, so a length 3 chain forces them to test that (and chain validation should be fairly automatic and offloaded to a crypto lib, so it wouldn't complicate health card implementer's code).

Reply:
Fair enough!